couchbase-entity/src/test/resources/EntityWithDocId.kt

```diff
@@ -0,0 +1,15 @@
+import kaufland.com.coachbasebinderapi.*
+import java.lang.String
+
+
+@Entity(database = "mydb_db")
+@Fields(
+        Field(name = "name", type = String::class),
+        Field(name = "type", type = String::class, defaultValue = "entityWithQueries", readonly = true),
+        Field(name = "identifiers", type = String::class, list = true),
+        Field(name = "map", type = Map::class, list = true)
+)
+@DocId("my:%name%:%type%")
```

Comment:
For certain document types we need custom code, e.g. to format ISO dates without the colon, is there a way to do this with the new annotation?

Reply:
I added a DocIdSegment annotation which allows you to declare methods for custom parts in a id. This annotation also allows you to depend on entity fields including subentities.

Hope that fits all requirements.